### PR TITLE
CR-1052320 XRT - installed and running CMC version is not printed out

### PR DESCRIPTION
**Pull request by sadasivanshaiju**
"platforms": [
                {
                    "static_region": {
                        "vbnv": "xilinx_u200_xdma_201830_2",
                        "logic_uuid": "0x5d1211e8",
                        "jtag_idcode": "0x14b37093",
                        "fpga_name": "xcu200-fsgd2104-2-e"
                    },
                    "off_chip_board_info": {
                        "ddr_size_bytes": "68719476736",
                        "ddr_count": "4"
                    },
                    "status": {
                        "mig_calibrated": "true",
                        "p2p_status": "disabled"
                    },
                    "controller": {
                        "satellite_controller": {
                            "version": "4.2.0",
                            "expected_version": "4.2.0"
                        },
                        "card_mgmt_controller": {
                            "version": "30.207.35",
                            "serial_number": "21290491E03K",
                            "oem_id": "N\/A"
                        }
                    },

### CODE REVIEW

File: src/runtime_src/core/common/utils.cpp [hunk 1]
```diff
@@ -26,6 +26,32 @@
 #include <limits>
 #include <boost/algorithm/string.hpp>
 
+std::string convert_version(std::stringstream& version)
```
Comment (stsoe) [inline src/runtime_src/core/common/utils.cpp:29]: Convert version to what?  The name does not explain much and I can't follow the logic of the function.
```
return-type
function_name(...)
{
}
```
Looks like the function should be static and within unnamed namespace.

File: src/runtime_src/core/common/info_platform.cpp [hunk 2]
```diff
@@ -140,10 +140,12 @@ add_controller_info(const xrt_core::device* device, ptree_type& pt)
     std::stringstream version;
     
     try {
-       version << "0x" << std::hex << std::stoi(xrt_core::device_query<xq::xmc_version>(device));
+       version << std::hex << std::stoi(xrt_core::device_query<xq::xmc_version>(device));
```
Comment (stsoe) [inline src/runtime_src/core/common/info_platform.cpp:143]: Why is version a stringstream?

File: src/runtime_src/core/common/utils.h [hunk 3]
```diff
@@ -34,8 +34,11 @@
 #define CU_AP_CONTINUE	(0x1 << 4)
 #define CU_AP_RESET	(0x1 << 5)
 
+std::string convert_version(std::stringstream& version);
```
Comment (stsoe) [inline src/runtime_src/core/common/utils.h:37]: Remove.  This function is not used outside of utils.cpp.

**Review by stsoe — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/common/utils.cpp (hunk 1 above)
Comment (sadasivanshaiju) [reply src/runtime_src/core/common/utils.cpp:29]: This function  is  called  in src/runtime_src/core/common/info_platform.cpp .  So it  can't be static

File: src/runtime_src/core/common/utils.h (hunk 3 above)
Comment (sadasivanshaiju) [reply src/runtime_src/core/common/utils.h:37]: This  function is  called in  src/runtime_src/core/common/info_platform.cpp  .  the above  header file is  included in src/runtime_src/core/common/info_platform.cpp .  So  this  declaration is needed

File: src/runtime_src/core/common/utils.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (sadasivanshaiju) [reply src/runtime_src/core/common/utils.cpp:29]: Please refer the  below CR  for  explanation of  the problem and  solution.

Change RequestCR-1052320
XRT - installed and running CMC version is not printed out

File: src/runtime_src/core/common/info_platform.cpp [hunk 4]
```diff
@@ -140,10 +140,12 @@ add_controller_info(const xrt_core::device* device, ptree_type& pt)
...
     }
     catch (...) {}
-    cmc.add("version", version.str());
+
+    std::string version_str = convert_version(version);
```
[the remark this answers is not in the document]
Comment (sadasivanshaiju) [reply src/runtime_src/core/common/info_platform.cpp:147]: Please  refer  the  below  CR  for  the example of  the problem and solution
Change RequestCR-1052320
XRT - installed and running CMC version is not printed out

File: src/runtime_src/core/common/utils.h (hunk 3 above)
Comment (stsoe) [inline src/runtime_src/core/common/utils.h:37]: How is this function to be used?  You say it is used from multiple locations, but I have no idea what "current_version" means, what the function returns, or what the argument is.    You can't just add a function in a header file without commenting what it is about.  

Lastly the function cannot be in the global namespace.

**Review by stsoe — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/common/info_platform.cpp (hunk 2 above)
Comment (sadasivanshaiju) [reply src/runtime_src/core/common/info_platform.cpp:143]: std::hex  need stringstream

Comment (stsoe) [reply src/runtime_src/core/common/info_platform.cpp:143]: No it does not.

**Comment by dbenusov-xilinx** [comment]
This PR was completed and handled here: https://github.com/Xilinx/XRT/pull/6078
Closing this to keep things neat!

**Pull request closed without merging**